dmoj/graders/interactive.py

```diff
@@ -0,0 +1,75 @@
+from dmoj.graders.standard import StandardGrader
+
+class WrongAnswer(BaseException):
+    pass
+
+class Interactor(object):
+    def __init__(self, process):
+        self.process = process
+        self._tokens = None
+
+    def read(self):
+        return self.process.stdout.read()
+
+    def readln(self):
+        return self.process.stdout.readline().rstrip()
+
+    def readtoken(self, delim=None):
+        if not self._tokens:
+            self._tokens = self.readln()
+        try:
+            ret, self._tokens = self._tokens.split(delim, 1)
+        except ValueError:
+            ret = self._tokens
+            self._tokens = None
+        return ret
+
+    def readint(self, lo=float('-inf'), hi=float('inf'), delim=None):
+        try:
+            ret = int(self.readtoken(delim))
+        except ValueError:
+            raise WrongAnswer
+        if not lo < ret < hi:
```

Comment:
[That's what Tudor said](https://github.com/DMOJ/judge/pull/244#discussion_r105577740)